api/v1alpha1/multiclusterservice_types.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MultiClusterService is the Schema for the multiclusterservice API
+type MultiClusterService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MultiClusterServiceSpec   `json:"spec,omitempty"`
+	Status MultiClusterServiceStatus `json:"status,omitempty"`
+}
+
+// MultiClusterServiceSpec defines the desired state of MultiClusterService
+type MultiClusterServiceSpec struct {
+	// ClusterSelector identifies target clusters to manage services on.
+	// +optional
+	ClusterSelector metav1.LabelSelector `json:"clusterSelector,omitempty"`
+	// Tier sets the priority for the services defined in this spec.
+	// Higher value means lower priority and vice versa.
+	// In case of conflict with another object managing the service,
+	// the one with lower priority will get to deploy its services.
+	// +kubebuilder:default:=100
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	Tier int32 `json:"tier,omitempty"`
+	// StopOnConflict specifies what to do in case of a conflict.
+	// E.g. If another object is already managing a service.
+	// By default the remaining services will be deployed even if conflict is detected.
+	// If set to true, the deployment will stop after encountering the first conflict.
+	// +optional
+	StopOnConflict bool `json:"stopOnConflict,omitempty"`
```

Comment:
what do you think if we align the name (here and in the MC) with the sveltos `profiles` `continueOnConflict`? Or would it be rather confusing in such a case?

documentation wise, may it also be a good idea to refer to the `github.com/projectsveltos/addon-controller.Spec` as a reference to the `tier`/`onconflict` fields?

Reply:
> what do you think if we align the name (here and in the MC) with the sveltos profiles continueOnConflict? Or would it be rather confusing in such a case?

I think we can use `continueOnConflict`. The reason I used `stopOnConflict` is because my assumption is that we would want remaining services to continue to be deployed if there is a conflict. So with `stopOnConflict`, we wouldn't have to set it in most cases as it will be `false` by default. However, I'll ask on slack whether we want the default behaviour to be to stop or continue on conflict.

Reply:
> documentation wise, may it also be a good idea to refer to the github.com/projectsveltos/addon-controller.Spec as a reference to the tier/onconflict fields?

We want to keep our API as Sveltos agnostic as possible so that's why I didn't refer.